src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1508,20 +1508,20 @@ private function registerSerializerConfiguration(array $config, ContainerBuilder
         };
 
         foreach ($container->getParameter('kernel.bundles_metadata') as $bundle) {
-            $dirname = $bundle['path'];
+            $configDir = isset($bundle['dir']) && is_dir($bundle['dir'].'/config') ? $bundle['dir'].'/config' : $bundle['path'].'/Resources/config';
```

Comment:
The legacy dirs should always taken into account. If a bundle used `config` directory in this case for something totally different, the actual `/Resources/config` will be skipped. So an existing bundle stops working just by having a certain directory.

Reply:
Yes, I agree that legacy dirs should have priority. I'll make the changes, thanks!